extensions/blocks/donations/donations.php

```diff
@@ -40,29 +94,170 @@ function register_block() {
  * @return string
  */
 function render_block( $attr, $content ) {
+	// Keep content as-is if rendered in other contexts than frontend (i.e. feed, emails, API, etc.).
+	if ( ! jetpack_is_frontend() ) {
+		return $content;
+	}
+
 	Jetpack_Gutenberg::load_assets_as_required( FEATURE_NAME, array( 'thickbox' ) );
 
-	require_once JETPACK__PLUGIN_DIR . '/modules/memberships/class-jetpack-memberships.php';
+	require_once JETPACK__PLUGIN_DIR . 'modules/memberships/class-jetpack-memberships.php';
+	require_once JETPACK__PLUGIN_DIR . 'class-jetpack-currencies.php';
+
 	add_thickbox();
 
 	$donations = array(
-		'one-time' => $attr['oneTimeDonation'],
-		'monthly'  => $attr['monthlyDonation'],
-		'annual'   => $attr['annualDonation'],
+		'one-time' => array_merge(
+			array(
+				'title' => __( 'One-Time', 'jetpack' ),
+				'class' => 'donations__one-time-item',
+			),
+			$attr['oneTimeDonation']
+		),
 	);
+	if ( $attr['monthlyDonation']['show'] ) {
+		$donations['1 month'] = array_merge(
+			array(
+				'title' => __( 'Monthly', 'jetpack' ),
+				'class' => 'donations__monthly-item',
+			),
+			$attr['monthlyDonation']
+		);
+	}
+	if ( $attr['annualDonation']['show'] ) {
+		$donations['1 year'] = array_merge(
+			array(
+				'title' => __( 'Yearly', 'jetpack' ),
+				'class' => 'donations__annual-item',
+			),
+			$attr['annualDonation']
+		);
+	}
+
+	$currency = $attr['currency'];
+
+	$classes = 'wp-block-jetpack-donations';
+	if ( ! empty( $attr['className'] ) ) {
+		$classes .= ' ' . $attr['className'];
+	}
+
+	$nav        = '';
+	$headings   = '';
+	$amounts    = '';
+	$extra_text = '';
+	$buttons    = '';
 	foreach ( $donations as $interval => $donation ) {
-		if ( ! $donation['show'] ) {
-			continue;
-		}
 		$plan_id = (int) $donation['planId'];
 		$plan    = get_post( $plan_id );
 		if ( ! $plan || is_wp_error( $plan ) ) {
 			continue;
 		}
 
-		$url     = \Jetpack_Memberships::get_instance()->get_subscription_url( $plan_id );
-		$content = preg_replace( '/(donations__donate-button donations__' . $interval . '-item")/i', '$1 href="' . esc_url( $url ) . '"', $content );
+		if ( count( $donations ) > 1 ) {
+			if ( ! $nav ) {
+				$nav .= '<div class="donations__nav">';
+			}
+			$nav .= sprintf(
+				'<div role="button" tabindex="0" class="donations__nav-item" data-interval="%1$s">%2$s</div>',
+				esc_attr( $interval ),
+				$donation['title']
```

Comment:
Can we escape this as well?

If the button is meant to accept some HTML (line breaks, bold, italic, and such), maybe we can rely on something like `wp_kses_post()`?

Reply:
> If the button is meant to accept some HTML (line breaks, bold, italic, and such), maybe we can rely on something like wp_kses_post()

This specific button is not expected to contain any HTML, I simply forgot to escape it with `esc_html` 🙂. However, I didn't escape the other parts you mentioned below for what you said, they can have some HTML but didn't think of using `wp_kses_post`. 

Updated all of them in 23cd23b7dc60dc74f82222ffd35a1e230a8e7ee5